### PR TITLE
Run rustfmt on selectors, servo_arc, and style

### PR DESCRIPTION
**Pull request by bholley**
Now that rustfmt is getting close to stable, and work on the style system has died down a bit, it seemed like an opportune time to auto-format the style crates.

The first commit disables import reordering, since tidy and rustfmt don't currently agree on the correct ordering. The second commit does a bunch of manual fixups such that the output of rustfmt passes tidy. The third commit runs rustfmt on the three aforementioned crate.

There are a few dozen warnings in the style crate about lines longer than 100 characters. It would be good to fix these, but I don't have time for that now.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/20603)

### CODE REVIEW

**Comment by bholley** [comment]
CC @upsuper @emilio @SimonSapin @pcwalton @nox @nrc

**Comment by bholley** [comment]
(and @heycam of course)

r? @SimonSapin

**Comment by bholley** [comment]
CC @Manishearth also

**Comment by upsuper** [comment]
I wonder is this really just running rustfmt on the crates? It seems to be doing more things than I would thought a formatter normally does, e.g. removing `use` for those only used once.

Also I find it a bit hard to follow how some decision is made for formatting, for example, I see conversion from single line `if ... { A } else { B }` to multi-line, and vice versa.

**Comment by upsuper** [comment]
Hmmm, actually not. Many of the changes I wouldn't expect formatter to do is from the manual fixup. That explains :)

File: components/selectors/parser.rs [hunk 1]
```diff
@@ -1901,33 +1904,32 @@ where
...
                     if P::parse_slotted(parser) && name.eq_ignore_ascii_case("slotted") {
-                        SimpleSelectorParseResult::SlottedPseudo(
-                            input.parse_nested_block(|input| {
-                                parse_inner_compound_selector(
-                                    parser,
-                                    input,
-                                )
-                            })?
-                        )
+                        let r = input.parse_nested_block(|input| {
```
Comment (emilio) [inline components/selectors/parser.rs]: what does `r` mean? Maybe just `selector`?

**Comment by emilio** [comment]
CI is also red:

```
error: unused import: `malloc_size_of::MallocUnconditionalShallowSizeOf`
  --> components/style/stylesheets/style_rule.rs:10:5
   |
10 | use malloc_size_of::MallocUnconditionalShallowSizeOf;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `-D unused-imports` implied by `-D warnings`
```

**Comment by SimonSapin** [comment]
@bholley Could you include in the PR message (which will go in the merge commit) what version of rustfmt you used and what command you ran to produce this commit?

**Comment by bholley** [comment]
Addressed review comments. If someone could stamp this soon I'd appreciate it, since it bitrots quickly. :-)

File: components/selectors/matching.rs [hunk 2]
```diff
@@ -267,12 +260,12 @@ where
...
-                selector.combinator_at_parse_order(from_offset),
-                Combinator::SlotAssignment | Combinator::PseudoElement
-            )
-        ),
+        iter.clone().next().is_some() ||
+            (from_offset != selector.len() &&
+                matches!(
+                    selector.combinator_at_parse_order(from_offset),
+                    Combinator::SlotAssignment | Combinator::PseudoElement
+                )),
```
Comment (emilio) [inline components/selectors/matching.rs]: Don't understand how this makes more sense than what we had either but.. I guess.

File: components/selectors/matching.rs [hunk 3]
```diff
@@ -378,36 +364,33 @@ fn matches_hover_and_active_quirk<Impl: SelectorImpl>(
...
-            Component::NthLastOfType(_, _) |
-            Component::FirstOfType |
-            Component::LastOfType |
-            Component::OnlyOfType => false,
-            Component::NonTSPseudoClass(ref pseudo_class) => {
-                pseudo_class.is_active_or_hover()
-            },
-            _ => true,
-        }
+    let all_match = selector_iter.clone().all(|simple| match *simple {
```
Comment (emilio) [inline components/selectors/matching.rs]: Ugh, this one looks really bad :(

File: components/selectors/matching.rs [hunk 4]
```diff
@@ -772,105 +737,80 @@ where
...
-                    matches_complex_selector(
-                        selector.iter(),
-                        element,
-                        context,
-                        flags_setter,
-                    )
+            context
+                .shared
+                .shadow_host()
+                .map_or(false, |host| host == element.opaque()) &&
```
Comment (emilio) [inline components/selectors/matching.rs]: Ugh, wtf. This were two conditions before and now looks a bit messy...

File: components/selectors/parser.rs [hunk 5]
```diff
@@ -727,11 +756,11 @@ impl Combinator {
     /// Returns true if this combinator is a child or descendant combinator.
     #[inline]
     pub fn is_ancestor(&self) -> bool {
```
Comment (emilio) [inline components/selectors/parser.rs]: This function looks also pretty bad...

File: components/style/author_styles.rs [hunk 6]
```diff
@@ -57,27 +56,20 @@ where
...
         }
 
         // Ignore OOM.
-        let _ = self.data.rebuild(
-            device,
-            quirks_mode,
-            flusher.sheets,
-            guard,
-        );
+        let _ = self.data
```
Comment (emilio) [inline components/style/author_styles.rs]: This looks like a regression too.

File: components/style/author_styles.rs [hunk 7]
```diff
@@ -57,27 +56,20 @@ where
...
-    where
+    ) where
         E: TElement,
         S: ToMediaListKey,
     {
-        let flusher = self.stylesheets.flush::<E>(
-            /* host = */ None,
-            /* snapshot_map = */ None,
-        );
+        let flusher = self.stylesheets
```
Comment (emilio) [inline components/style/author_styles.rs]: This looks worse imo.

File: components/style/counter_style/mod.rs [hunk 8]
```diff
@@ -149,7 +156,7 @@ impl<'a, 'b, 'i> AtRuleParser<'i> for CounterStyleRuleParser<'a, 'b> {
 
 macro_rules! checker {
     ($self:ident._($value:ident)) => {};
-    ($self:ident.$checker:ident($value:ident)) => {
+    ($self:ident. $checker:ident($value:ident)) => {
```
Comment (emilio) [inline components/style/counter_style/mod.rs]: This is touching a macro, and I'm not sure it's fine, actually... I guess it is but it looks fishy at best.

File: components/style/dom.rs [hunk 9]
```diff
@@ -338,15 +345,8 @@ pub trait TShadowRoot : Sized + Copy + Clone + PartialEq {
...
-    : Eq
-    + PartialEq
-    + Debug
-    + Hash
-    + Sized
-    + Copy
-    + Clone
-    + SelectorsElement<Impl = SelectorImpl>
+pub trait TElement:
+    Eq + PartialEq + Debug + Hash + Sized + Copy + Clone + SelectorsElement<Impl = SelectorImpl>
```
Comment (emilio) [inline components/style/dom.rs]: This looks also like a regression to me, but...

File: components/style/error_reporting.rs [hunk 10]
```diff
@@ -96,88 +104,107 @@ impl<'a> fmt::Display for ContextualParseError<'a> {
...
-                write!(f, "Invalid @counter-style rule: 'system: additive' without 'additive-symbols'")
-            }
-            ContextualParseError::InvalidCounterStyleExtendsWithSymbols => {
-                write!(f, "Invalid @counter-style rule: 'system: extends …' with 'symbols'")
-            }
-            ContextualParseError::InvalidCounterStyleExtendsWithAdditiveSymbols => {
-                write!(f, "Invalid @counter-style rule: 'system: extends …' with 'additive-symbols'")
-            }
+            },
+            ContextualParseError::InvalidCounterStyleWithoutSymbols(ref system) => write!(
```
Comment (emilio) [inline components/style/error_reporting.rs]: These look way worse too.

File: components/style/values/specified/outline.rs [hunk 11]
```diff
@@ -10,8 +10,8 @@ use selectors::parser::SelectorParseErrorKind;
 use style_traits::ParseError;
 use values::specified::BorderStyle;
 
-#[derive(Clone, Copy, Debug, Eq, MallocSizeOf, Ord)]
-#[derive(PartialEq, PartialOrd, ToComputedValue, ToCss)]
+#[derive(Clone, Copy, Debug, Eq, MallocSizeOf, Ord, PartialEq, PartialOrd, ToComputedValue,
+         ToCss)]
```
Comment (emilio) [inline components/style/values/specified/outline.rs]: It also merges `#[derive(..)]` lines? Is there a way to prevent it from doing that?

**Review by Manishearth — APPROVED** [review]
r=me

File: components/selectors/builder.rs [hunk 12]
```diff
@@ -266,28 +262,28 @@ impl From<u32> for Specificity {
...
+where
+    Impl: SelectorImpl,
 {
     fn simple_selector_specificity<Impl>(
         simple_selector: &Component<Impl>,
         specificity: &mut Specificity,
-    )
-    where
-        Impl: SelectorImpl
+    ) where
```
Comment (emilio) [inline components/selectors/builder.rs]: This (`) where`) looks weird, is it expected?

**Review by emilio — APPROVED** [review]
There are a few of these that look like regressions to me. Maybe @nrc may take a look and see if there's something unexpected in some of these?

Also, is there any plan / way to enforce its usage? My plan to start doing this, btw, was enforcing usage of `rustfmt-format-diff` (https://github.com/rust-lang-nursery/rustfmt/blob/master/src/format-diff/main.rs) on incoming commits. But not sure if there's any better idea around.

Anyway, as much as I dislike some particular changes, consistency is king I guess...

**Comment by emilio** [comment]
FWIW I think something like #20617 would be less invasive. Arguably needs more polish and what not. But massive reformats pollute the history quite heavily, it's better to do it incrementally if possible IMO.

**Comment by bholley** [comment]
The downsides I see to per-patch reformatting are:
(1) Patches that should otherwise be small (i.e. renames) can suddenly get big as they reformat surrounding code, making them harder to review.
(2) It's harder for new contributors to infer the correct style from surrounding code, because it may or may not be consistent with the canonical style.

There's definitely room for disagreement here. But given that most people have expressed approval for doing it this way, and given that I've already done the work here, I'm inclined to land it.
